### PR TITLE
Add `LocationTag.structure_block_data` tag + mech pair

### PR DESCRIPTION
**Pull request by tal5**
## Additions

- `LocationTag.structure_block_data` Tag + Mech pair - Controls the structure block data of the structure block at the location.

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java [hunk 1]
```diff
@@ -4230,6 +4233,55 @@ else if (material.hasModernData() && material.getModernData() instanceof org.buk
...
+        // -->
+        tagProcessor.registerTag(MapTag.class, "structure_block_data", (attribute, object) -> {
+            BlockState state = object.getBlockStateForTag(attribute);
+            if (!(state instanceof Structure)) {
+                attribute.echoError("Location is not a valid Structure block.");
+                return null;
+            }
+            Structure structure = (Structure) state;
+            MapTag output = new MapTag();
+
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java]: stray newlines

File: plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java [hunk 2]
```diff
@@ -5116,6 +5168,186 @@ else if (state instanceof Dropper) {
...
+        // @tags
+        // <LocationTag.structure_block_data>
+        // -->
+        if (mechanism.matches("structure_block_data") && mechanism.requireObject(MapTag.class)) {
+            BlockState state = getBlockState();
+            if (!(state instanceof Structure)) {
+                mechanism.echoError("'structure_block_data' mechanism can only be called on Structure blocks.");
+                return;
+            }
+
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java]: stray newlines

File: plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java (hunk 1 above)
Comment (tal5) [reply plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java]: Should be fixed

File: plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java (hunk 2 above)
Comment (tal5) [reply plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java]: Should be fixed

**Pull request merged**